proposals/Fleet-Windows-OS-vulns.md

```diff
@@ -0,0 +1,139 @@
+the CISA list of know vulnerabilities as a sample, it was determined that all vulnerability
+remediations follow the form of *applying some software patch* and then *following some
+steps (if any)* so it follows that to determine if a system is susceptible to a vulnerability we will need to
+check whether a specific software patch was applied and also if the provided steps were followed. So
+basically we have three problems we need to solve: 
+1. We need a list of vulnerabilities + remediations. Each remediation is composed of a patch + some steps.
+2. For a given vulnerability we need to determine whether the proper patch was applied.
+3. For a given vulnerability we need to determine whether the steps (if any) were followed.
+
+## List of vulnerabilities
+
+To get the list of vulnerabilities we can use the [Microsoft Security Updates API](https://api.msrc.microsoft.com/cvrf/v2.0/swagger/index). This [endpoint](api.msrc.microsoft.com/cvrf/v2.0/document/yyyy-mm) in particular exposes all the security updates released for yyyy-mmm formatted according to the [Common Vulnerability Reporting Framework](http://docs.oasis-open.org/csaf/csaf-cvrf/v1.2/csaf-cvrf-v1.2.html#:~:text=The%20CSAF%20Common%20Vulnerability%20Reporting,impact%20and%20remediation%20among%20interested) (cvrf) format, this cvrf document will include entries for vulnerabilities, how to patch them and what products are affected. 
+
+For example, say we have a host running *Windows 10 Version 21H2 x64* and we want to determine whether that host is susceptible to *CVE-2022-26925*. Looking at https://api.msrc.microsoft.com/cvrf/v2.0/document/2022-May we can see that  *Windows 10 Version 21H2 for x64* is assigned the Product ID **11931**:
+
+```
+...
+<prod:ProductTree>
+    ...
+    <prod:Branch Type="Product Family" Name="Windows">
+    ...
+        <prod:FullProductName ProductID="11931">Windows 10 Version 21H2 for x64-based Systems</prod:FullProductName>
+        ...
+    </prod:Branch>
+    …
+<prod:ProductTree>
+…
+```
+
+Looking at the matching vulnerability element (`<vuln:Vulnerability Ordinal="47">`) *CVE-2022-26925*, we can see that
+*Windows 10 Version 21H2 x64* is indeed affected by it:
+
+```
+<vuln:ProductStatuses>
+    <vuln:Status Type="Known Affected">
+    ...
+        <vuln:ProductID>11923</vuln:ProductID>
+    ...
+    </vuln:Status>
+</vuln:ProductStatuses>
+```
+
+And also the remediation steps:
+
+```
+<vuln:Remediations>
+…
+    <vuln:Remediation Type="Vendor Fix">
+        <vuln:Description>5013942</vuln:Description>
+        <vuln:URL>https://catalog.update.microsoft.com/v7/site/Search.aspx?q=KB5013942</vuln:URL>
+        <vuln:Supercedence>5012599</vuln:Supercedence>
+        ..
+        <vuln:ProductID>11931</vuln:ProductID>
+        ...
+        <vuln:RestartRequired>Yes</vuln:RestartRequired>
+        …
+    </vuln:Remediation>
+    <vuln:Remediation Type="Known Issue">
```

Comment:
Why would there be multiple known issue acknowledgements for a single issue?

Reply:
A single vulnerability can affect different products, each one of these unique (vulnerability, products) pairs can have different 'known issue' article. What wouldn't make (at least from my point of view) is having multiple 'known issue' articles for the same (vulnerability, products) pair.

Reply:
I see. thank you!